### PR TITLE
OBPIH-5501 replace translated name with display name in stock movements, returns and combined shipment

### PR DESCRIPTION
**Pull request by drodzewicz**

### CODE REVIEW

File: src/js/components/returns/inbound/AddItemsPage.jsx [hunk 1]
```diff
@@ -260,14 +260,9 @@ class AddItemsPage extends Component {
...
           const inboundReturn = parseResponse(resp.data.data);
-          const returnItems = inboundReturn.stockTransferItems.length > 0 ?
-            _.map(inboundReturn.stockTransferItems, item => ({
-              ...item,
-              product: {
-                ...item.product,
-                label: `${item.product.productCode} ${item.product.translatedName ?? item.product.name}`,
-              },
-            })) : new Array(1).fill({ sortOrder: 100 });
+          const returnItems = inboundReturn.stockTransferItems.length > 0
```
Comment (kchelstowski) [inline src/js/components/returns/inbound/AddItemsPage.jsx:263]: I know it's not your code, but if we already "touch" some part of the code that could be simplified, just a short suggestion, which I'm sure you are aware of, but just didn't see it
```suggestion
          const returnItems = inboundReturn.stockTransferItems.length
```

File: src/js/components/stock-movement-wizard/combined-shipments/AddItemsPage.jsx [hunk 2]
```diff
@@ -314,10 +311,6 @@ class AddItemsPage extends Component {
       val => ({
         ...val,
         disabled: true,
-        product: {
-          ...val.product,
-          label: `${val.productCode} ${val.product.translatedName || val.product.name}`,
-        },
```
Comment (kchelstowski) [inline src/js/components/stock-movement-wizard/combined-shipments/AddItemsPage.jsx:320]: :+1: for catching all of those

File: src/js/utils/option-utils.jsx [hunk 3]
```diff
@@ -114,8 +114,8 @@ export const debounceAvailableItemsFetch = (waitTime, minSearchLength) =>
             quantityAvailable: obj.quantityAvailable,
             minExpirationDate: obj.minExpirationDate,
             handlingIcons: obj.product.handlingIcons,
-            label: `${obj.productCode} - ${obj.translatedName ?? obj.name}`,
-            hasTranslatedName: !!obj.translatedName,
+            label: `${obj.productCode} - ${obj?.product.displayNames?.default ?? obj.name}`,
+            hasTranslatedName: !!obj?.product?.displayNames?.default,
```
Comment (kchelstowski) [inline src/js/utils/option-utils.jsx:118]: shouldn't it be renamed to:
```suggestion
            hasDisplayName: !!obj?.product?.displayNames?.default,
```

or even as we've discussed in my pr, just keep:
```javascript
displayName: obj?.product?.displayNames?.default,
```

File: src/js/utils/option-utils.jsx [hunk 4]
```diff
@@ -136,8 +136,8 @@ export const debounceProductsInOrders = (waitTime, minSearchLength, vendor, dest
             name: obj.name,
             productCode: obj.productCode,
             handlingIcons: obj.handlingIcons,
-            label: `${obj.productCode} - ${obj.translatedName ?? obj.name}`,
-            hasTranslatedName: !!obj.translatedName,
+            label: `${obj.productCode} - ${obj.displayNames?.default ?? obj.name}`,
+            hasTranslatedName: !!obj.displayNames?.default,
```
Comment (kchelstowski) [inline src/js/utils/option-utils.jsx:140]: same as above:
```suggestion
           displayName: obj.displayNames?.default,
```

**Review by kchelstowski — CHANGES_REQUESTED** [review]

**Review by awalkowiak — APPROVED** [review]
Overall LGTM, please take a look at Kacper's suggestions.

File: src/js/utils/option-utils.jsx (hunk 4 above)
Comment (drodzewicz) [reply src/js/utils/option-utils.jsx:140]: From what I can see, it appears to be that `hasTranslatedName` from **debounceAvailableItemsFetch** and **debounceProductsInOrders** is not being used anywhere, so I think it's safe to remove it.

File: src/js/components/product-select/ProductSelect.jsx [hunk 5]
```diff
@@ -8,15 +8,19 @@ import { debounceProductsFetch } from 'utils/option-utils';
 import renderHandlingIcons from 'utils/product-handling-icons';
 import Select from 'utils/Select';
 
+const productOptionLabel = ({
+  productCode, displayName, displayNames, name,
+}) => (`${productCode} - ${displayName || displayNames?.default || name}`);
+
```
Comment (kchelstowski) [inline src/js/components/product-select/ProductSelect.jsx:14]: :+1:

**Review by kchelstowski — APPROVED** [review]

**Pull request merged**